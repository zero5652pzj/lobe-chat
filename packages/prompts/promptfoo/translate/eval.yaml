description: Test translation accuracy between different languages

providers:
  - openai:chat:gpt-5-mini
  - openai:chat:claude-3-5-haiku-latest
  - openai:chat:gemini-flash-latest
  - openai:chat:deepseek-chat

prompts:
  - file://promptfoo/translate/prompt.ts

tests:
  - vars:
      content: 'Hello, how are you?'
      from: 'en-US'
      to: 'zh-CN'
    assert:
      - type: contains-any
        value: ['你好', '您好']
      - type: not-contains
        value: 'Hello'

  - vars:
      content: '你好，你怎么样？'
      from: 'zh-CN'
      to: 'en-US'
    assert:
      - type: contains-any
        value: ['Hello', 'Hi', 'how are you', 'How are you']
      - type: not-contains
        value: '你好'

  - vars:
      content: 'Je suis content de vous rencontrer'
      from: 'fr-FR'
      to: 'en-US'
    assert:
      - type: contains-any
        value: ['pleased', 'happy', 'glad', 'meet', 'meeting']
      - type: not-contains
        value: 'Je suis'

  - vars:
      content: 'The weather is beautiful today'
      from: 'en-US'
      to: 'es-ES'
    assert:
      - type: contains-any
        value: ['tiempo', 'clima', 'hermoso', 'bonito', 'hoy', 'día', 'precioso']
      - type: not-contains
        value: 'weather'

  - vars:
      content: 'I love programming with TypeScript'
      from: 'en-US'
      to: 'ja-JP'
    assert:
      - type: contains
        value: 'TypeScript' # Technical terms often remain unchanged
      - type: not-contains
        value: 'I love'

  - vars:
      content: 'Machine learning is revolutionizing technology'
      from: 'en-US'
      to: 'de-DE'
    assert:
      - type: contains-any
        value: ['Technologie', 'revolutioniert', 'maschinelles', 'Lernen']
      - type: not-contains
        value: 'Machine learning'

  - vars:
      content: 'API_KEY_12345'
      from: 'en-US'
      to: 'zh-CN'
    assert:
      - type: contains
        value: 'API_KEY_12345'
