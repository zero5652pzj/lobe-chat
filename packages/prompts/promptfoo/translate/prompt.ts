// TypeScript prompt wrapper that uses actual chain implementation
import { chainTranslate } from '@lobechat/prompts';

interface PromptVars {
  content: string;
  from: string;
  to: string;
}

export default function generatePrompt({ vars }: { vars: PromptVars }) {
  const { content, to } = vars;

  // Use the actual chain function from src
  const result = chainTranslate(content, to);

  // Return messages array as expected by promptfoo
  return result.messages || [];
}
