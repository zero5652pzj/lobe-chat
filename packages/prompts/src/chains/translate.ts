import { ChatStreamPayload } from '@lobechat/types';

export const chainTranslate = (
  content: string,
  targetLang: string,
): Partial<ChatStreamPayload> => ({
  messages: [
    {
      content: `You are a professional translator. Translate the input text to ${targetLang}.

Rules:
- Output ONLY the translated text, no explanations or additional context
- Preserve technical terms, code identifiers, API keys, and proper nouns exactly as they appear
- Maintain the original formatting and structure
- Use natural, idiomatic expressions in the target language`,
      role: 'system',
    },
    {
      content,
      role: 'user',
    },
  ],
});
