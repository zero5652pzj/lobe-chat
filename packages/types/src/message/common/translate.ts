export interface Translate {
  from?: string;
  to: string;
}

export interface ChatTranslate extends Translate {
  content?: string;
}
