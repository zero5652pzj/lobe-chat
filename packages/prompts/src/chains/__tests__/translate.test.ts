import { describe, expect, it } from 'vitest';

import { chainTranslate } from '../translate';

describe('chainTranslate', () => {
  it('should create a payload with system and user messages for translation', () => {
    // Arrange
    const content = 'This is a test sentence.';
    const targetLang = 'zh-CN';

    // Act
    const result = chainTranslate(content, targetLang);

    // Assert
    expect(result).toMatchSnapshot();
  });
});
