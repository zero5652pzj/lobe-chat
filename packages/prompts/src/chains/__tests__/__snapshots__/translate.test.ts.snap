// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`chainTranslate > should create a payload with system and user messages for translation 1`] = `
{
  "messages": [
    {
      "content": "You are a professional translator. Translate the input text to zh-CN.

Rules:
- Output ONLY the translated text, no explanations or additional context
- Preserve technical terms, code identifiers, API keys, and proper nouns exactly as they appear
- Maintain the original formatting and structure
- Use natural, idiomatic expressions in the target language",
      "role": "system",
    },
    {
      "content": "This is a test sentence.",
      "role": "user",
    },
  ],
}
`;
